includes/class-wc-subscriptions-product.php

```diff
@@ -724,7 +724,7 @@ public static function maybe_set_variations_price_html( $variation_details, $var
 	public static function user_can_not_delete_subscription( $allcaps, $caps, $args ) {
 		global $wpdb;
 
-		if ( isset( $args[0] ) && in_array( $args[0], array( 'delete_post', 'delete_product' ) ) && isset( $args[2] ) && ( ! isset( $_GET['action'] ) || 'untrash' != $_GET['action'] ) && 0 === strpos( get_post_type( $args[2] ), 'product' ) ) {
+		if ( isset( $args[0] ) && in_array( $args[0], array( 'delete_post', 'delete_product' ), true ) && isset( $args[2] ) && ( ! isset( $_GET['action'] ) || 'untrash' !== $_GET['action'] ) && 0 === strpos( WC_Data_Store::load( 'subscription' )->get_order_type( $args[2] ), 'product' ) ) { // phpcs:ignore
```

Comment:
Following similar [suggestion](https://github.com/Automattic/woocommerce-subscriptions-core/pull/306#discussion_r1031932685), I tried `WC_Data_Store::load( 'product' )->get_order_type()` but it didn't work. It returns `null`. I'm not sure why.

Reply:
For safety, we can revert this change and leave it as is since products are staying in the Posts table.

--


> WC_Data_Store::load( 'product' )->get_order_type()

To answer this question, the product data store doesn't have a `get_order_type()` function.
We need to use `get_product_type()` but this current `strpos` won't work becuase `WC_Data_Store::load( 'product' )->get_product_type()` changes the product type. For instance, for posts with type `product_variation` this function returns 'variation' so we can't look for product types that include 'product'

As I mentioned before, we can revert this change and leave it as is since it will require some bigger logic changes.

Reply:
Thank you for explaining all that. It makes so much sense. Reverted in [16b9bc5](https://github.com/Automattic/woocommerce-subscriptions-core/pull/318/commits/16b9bc566fff34aa6cc2a90240d22b506f91f1f4).